jdaviz/configs/specviz2d/plugins/parsers.py

```diff
@@ -46,9 +46,15 @@ def spec2d_1d_parser(app, data_obj, data_label=None, show_in_viewer=True):
             flux = np.sum(data, header['DISPAXIS']-1)*u.Unit(header["BUNIT"])
         except ValueError:
             flux = u.Quantity(np.sum(data, header['DISPAXIS']-1))
-        step = (header["WAVEND"] - header["WAVSTART"])/len(flux)
-        spectral_axis = np.arange(header["WAVSTART"], header["WAVEND"],
-                                  step) * u.Unit("m")
+
+        if "WAVEND" in header and "WAVSTART" in header:
+            step = (header["WAVEND"] - header["WAVSTART"])/len(flux)
+            spectral_axis = np.arange(header["WAVSTART"], header["WAVEND"],
+                                      step) * u.Unit("m")
+        else:
+            # u.Unit("m") is used if WAVEND and WAVSTART are present so
+            # we use it here as well, even though the actual unit is pixels
+            spectral_axis = np.arange(1, len(flux)+1, 1) * u.Unit("m")
```

Comment:
I am not sure how I feel about casting meter when you know it is in pixels, but I am no domain expert.
```suggestion
            spectral_axis = np.arange(1, flux.size + 1, 1) * u.m
```